test/e2e/aws_test.go

```diff
@@ -132,13 +139,113 @@ var _ = Describe("functional tests", func() {
 			By("Scale the MachineDeployment up again")
 			scaleMachineDeployment(namespace, machineDeploymentName, scaleDownReplicas, initialReplicas)
 
+			By("Creating the LB service")
+			sess := getSession()
+			elbClient := elb.New(sess)
+			createLBService(namespace, clusterName, lbServiceName)
+			// This sleep is required to let the LB service up and running
+			time.Sleep(15 * time.Second)
+			verifyElbExists(elbClient, lbServiceName, true)
+
 			By("Deleting the Cluster")
 			deleteCluster(namespace, clusterName)
 
+			By("Checking provisioned LB deleted")
+			verifyElbExists(elbClient, lbServiceName, false)
 		})
 	})
 })
 
+func createLBService(namespace, clusterName, svcName string) {
+	cluster := &clusterv1.Cluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: namespace,
+			Name:      clusterName,
+		},
+	}
+	kubeConfigData, err := kubeconfig.FromSecret(kindClient, cluster)
+	Expect(err).NotTo(HaveOccurred())
+
+	restConfig, err := clientcmd.RESTConfigFromKubeConfig(kubeConfigData)
+	Expect(err).NotTo(HaveOccurred())
+
+	svcClient, err := crclient.New(restConfig, crclient.Options{})
+	Expect(err).NotTo(HaveOccurred())
+
+	lbSvc := corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: "default",
+			Name:      svcName,
+		},
+		Spec: corev1.ServiceSpec{
+			Type: corev1.ServiceTypeLoadBalancer,
+			Ports: []corev1.ServicePort{{
+				Port:     80,
+				Protocol: corev1.ProtocolTCP,
+			}},
+			Selector: map[string]string{
+				"app": "nginx",
+			},
+		},
+	}
+	Expect(svcClient.Create(context.TODO(), &lbSvc)).NotTo(HaveOccurred())
+	svcCreated := &corev1.Service{}
+	Eventually(
+		func() (bool, error) {
+			if err := svcClient.Get(context.TODO(), apimachinerytypes.NamespacedName{Namespace: "default", Name: svcName}, svcCreated); err != nil {
+				fmt.Fprintf(GinkgoWriter, "Error while listing services: %v\n", err)
+				return false, err
+			}
+			if svcCreated.Name == svcName {
+				fmt.Fprintf(GinkgoWriter, "Found the service: %s\n", svcName)
+				return true, nil
+			}
+			return false, nil
+		},
+		5*time.Minute, 15*time.Second,
+	).Should(BeTrue())
+}
+
+func verifyElbExists(elbClient *elb.ELB, svcName string, exists bool) {
```

Comment:
Suggestion: It is better to get the external ip of the service created and then use it to verify.

Reply:
ELBs can be describe using ELB name (in case of elb) or ELB ARN name in case of elbv2). It is not possible to list them using IP. Hence, need to describe all the ELBs and iterate over them to find the required one created.

Reply:
```
"LoadBalancerName": "a067b07bbec5f4339a403992f01e5df1",
"DNSName": "a067b07bbec5f4339a403992f01e5df1-1901108057.us-east-2.elb.amazonaws.com",
```
ELB name can be extracted by removing the last `xxxxxxxxxx.us-east-2.elb.amazonaws.com` from the DNS name

Reply:
Rather than getting the external IP,  added code to fetch the ingress host name from the service and then extracted ELB name out of it which later used to check the existence.